safe/gui/tools/minimum_needs/test/test_minimum_needs_tool.py

```diff
@@ -52,39 +56,58 @@ def tearDown(self):
             if os.path.exists(path):
                 os.remove(path)
 
-    # This test is expected to fail until we fix the minimum needs calculator
-    # in InaSAFE V4.
-    @unittest.expectedFailure
     def test_minimum_needs(self):
-        """Test behaviour of the minimum needs function.
-        """
+        """Test behaviour of the minimum needs function."""
         dialog = NeedsCalculatorDialog(PARENT)
-        layer = shapefile_path
+        layer = QgsVectorLayer(
+            shapefile_path,
+            os.path.basename(shapefile_path),
+            'ogr')
+        if layer.isValid():
+            layer = dialog.prepare_new_layer(layer)
         attribute = 'displaced'
-        new_layer = dialog.minimum_needs(layer, attribute)
-        assert new_layer is not None
-        attributes = {
-            'Drinking Water': 17500,
-            'Family Kits': 200,
-            'Rice': 2800,
-            'Toilets': 50,
-            'Clean Water': 67000}
-        self.assertDictEqual(attributes, dict(new_layer.data[0]))
-
-    # def Xtest_accept(self):
-    #     """Test behaviour of the ok button.
-    #
-    #     TODO: Make this test useful - Tim
-    #     """
-    #     # print shapefile_path
-    #     layer = QgsVectorLayer(
-    #         os.path.basename(shapefile_path),
-    #         os.path.dirname(shapefile_path),
-    #         'ogr')
-    #     layer = None
-    #     dialog = MinimumNeeds(PARENT)
-    #     dialog.accept()
-
+        layer.keywords = {
+            'layer_purpose': layer_purpose_aggregation['key'],
+            'inasafe_fields': {displaced_field['key']: attribute}}
+        rename_remove_inasafe_fields(layer)
```

Comment:
The same here, I think you shouldn't call this function from the test.

The idea of writing test is to simulate the user, to write whet the user will do using the tool.
* Open the dialog, check that the run button is disabled (because there is no layer)
* Add a layer to the canvas, check that combobox are working well and the run button is enabled.
* Simulate the click on the run button
* Check the output layer.

In some other other words, we shouldn't call function which are inside the run such as `prepare_new_layer(layer)`, `minimum_needs(layer)` ...